src/messaging/ExchangeMgr.cpp

```diff
@@ -296,6 +298,50 @@ void ExchangeManager::OnMessageReceived(const PacketHeader & packetHeader, const
     }
 }
 
+ChannelHandle ExchangeManager::EstablishChannel(const ChannelBuilder & builder, ChannelDelegate * delegate)
+{
+    ChannelContext * channelContext = nullptr;
+
+    // Find an existing Channel matching the builder
+    mChannelContexts.ForEachActiveObject([&](ChannelContext * context) {
+        if (context->MatchesBuilder(builder, mSessionMgr))
+        {
+            channelContext = context;
+            return false;
+        }
+        return true;
+    });
+
+    if (channelContext == nullptr)
+    {
+        // create a new channel if not found
+        channelContext = mChannelContexts.CreateObject(this);
+        if (channelContext == nullptr)
+            return ChannelHandle{ nullptr };
+        channelContext->Start(builder);
+    }
+    else
+    {
+        channelContext->Retain();
+    }
+
+    auto association = mChannelHandles.CreateObject(channelContext, delegate);
```

Comment:
It's not obvious what type this creates ("Object" ?), can you spell it out instead of using auto?

Reply:
done

Reply:
Maybe this hints that 'CreateObject' is not ideal as a method name. 

Maybe we should change pool templates to just have 'Create' since them I would infer that `mChannleHandles` will "create a channel handle".